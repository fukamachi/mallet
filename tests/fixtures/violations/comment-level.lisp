(defpackage #:token-rules-example
  (:use #:cl))
(in-package #:token-rules-example)

;;; This file tests token-level rules (comment-level)

;; Bad: top-level comment should have 3 or 4 semicolons
(defun example1 ()
  ; Bad: inside function should have 2 semicolons
  (let ((x 1))
    (+ x 2)))

;;;; Good: file-level comment
;;; Good: section-level comment

(defun example2 ()
  ;; Good: line comment inside function
  (let ((y 2)) ; Good: inline comment
    (* y 3)))

; Bad: top-level should be 3 or 4 semicolons
(defvar *global* 42)
