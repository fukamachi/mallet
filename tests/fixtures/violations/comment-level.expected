# Expected violations for comment-level.lisp
# Format: line:column rule-name severity

7:0 comment-level warning
9:2 comment-level warning
22:0 comment-level warning
